servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java

```diff
@@ -1783,6 +1785,79 @@ public void testAssignRefToFreshMain() throws BaseNessieClientServerException {
     assertThat(testTagRef.getHash()).isEqualTo(main.getHash());
   }
 
+  @Test
+  public void testBranchesHaveMetadataProperties() throws BaseNessieClientServerException {
+    String branchPrefix = "branchesHaveMetadataProperties";
+    int numBranches = 5;
+    int commitsPerBranch = 10;
+
+    for (int i = 0; i < numBranches; i++) {
+      Reference r = api.createReference().reference(Branch.of(branchPrefix + i, null)).create();
+      String currentHash = r.getHash();
+      createCommits(r, 1, commitsPerBranch, currentHash);
+    }
+    // not fetching additional metadata
+    List<Reference> references = api.getAllReferences().get().getReferences();
+    Optional<Reference> main =
+        references.stream().filter(r -> r.getName().equals("main")).findFirst();
+    assertThat(main).isPresent();
+    List<Branch> branches =
+        references.stream()
+            .filter(r -> r.getName().startsWith(branchPrefix))
+            .map(r -> (Branch) r)
+            .collect(Collectors.toList());
```

Comment:
BTW: If you combine the two `assertThat(branches)` in both occurences, you can omit the `.collect(Collectors.toList())` and assert directly on the `Stream`

Reply:
good point, updated